tests/system/python/e2e/test_e2e_kafka.py

```diff
@@ -128,25 +128,41 @@ def test_end_to_end(self, start_south_north, foglamp_url, wait_time, kafka_host,
         assert 1 == len(val)
         assert {'sensor': SENSOR_VALUE} == val[0]["reading"]
 
-        self._read_from_kafka(kafka_host, kafka_rest_port, kafka_topic)
+        def get_ping_status():
+            _connection = http.client.HTTPConnection(foglamp_url)
+            _connection.request("GET", '/foglamp/ping')
+            cr = _connection.getresponse()
+            assert 200 == cr.status
+            res = cr.read().decode()
+            jdoc = json.loads(res)
+            assert len(jdoc)
+            assert 1 == jdoc['dataSent']
+            assert 1 == jdoc['dataRead']
 
-    def _read_from_kafka(self, host, rest_port, topic):
-        conn = http.client.HTTPConnection("{}:{}".format(host, rest_port))
+        get_ping_status()
+
+        self._read_from_kafka(kafka_host, kafka_rest_port, kafka_topic, wait_time)
+
+    def _clean_kafka_instance(self, kafka_host, kafka_rest_port):
+        """  Close the consumer (DELETE) to make it leave the group and clean up its resources
+        :param kafka_host:
+        :param kafka_rest_port:
+        :return:
+        """
+        self._close_consumer(kafka_host, kafka_rest_port)
 
-        # Close the consumer (DELETE) to make it leave the group and clean up its resources
-        self._close_consumer(host, rest_port)
+    def _read_from_kafka(self, host, rest_port, topic, wait_time):
 
+        self._clean_kafka_instance(host, rest_port)
+
+        conn = http.client.HTTPConnection("{}:{}".format(host, rest_port))
         # Assertions on Kafka topic, consumer and subscription
         self._verify_kafka_topic(conn, topic)
-
         self._verify_kafka_topic_by_name(conn, topic)
-
         # Create a consumer group and instance
         self._verify_kafka_consumer_group_and_instance(conn)
```

Comment:
with latest confluent kafka setup, there is a Assertion failure with 415 status code.